pcdsdevices/lodcm.py

```diff
@@ -84,21 +157,145 @@ class LODCM(BaseInterface, Device):
+
+    dh_prefix : str
+        LOM Dia H motor prefix.
+
+    dv_prefix : str
+        LOM Dia V motor prefix.
+
+    dr_prefix : str
+        LOM Dia Theta motor prefix.
+
+    df_prefix : str
+        LOM Dia Filter Wheel motor prefix.
+
+    dd_prefix : str
+        LOM Dia PIPS motor prefix.
+
+    yag_zoom_prefix : str
+        LOM Zoom motor prefix.
+    """
     h1n = Cpt(H1N, ':H1N', kind='hinted')
     yag = Cpt(YagLom, ":DV", kind='omitted')
     dectris = Cpt(Dectris, ":DH", kind='omitted')
     diode = Cpt(Diode, ":DIODE", kind='omitted')
     foil = Cpt(Foil, ":FOIL", kind='omitted')
+    y1_state = Cpt(Y1, ':Y1', kind='omitted')
+    chi1_state = Cpt(CHI1, ':CHI1', kind='omitted')
+    h2n_state = Cpt(H2N, ':H2N', kind='omitted')
+    y2_state = Cpt(Y2, ':Y2', kind='omitted')
+    chi2_state = Cpt(CHI2, ':CHI2', kind='omitted')
+
+    # Crystal Tower 1
+    z1 = UCpt(IMS, kind='normal', doc='LOM Xtal1 Z')
+    x1 = UCpt(IMS, kind='normal', doc='LOM Xtal1 X')
+    y1 = UCpt(IMS, kind='normal', doc='LOM Xtal1 Y')
+    th1 = UCpt(IMS, kind='normal', doc='LOM Xtal1 Theta')
+    ch1 = UCpt(IMS, kind='normal', doc='LOM Xtal1 Chi')
+    h1n_m = UCpt(IMS, kind='normal', doc='LOM Xtal1 Hn')
+    h1p = UCpt(IMS, kind='normal', doc='LOM Xtal1 Hp')
+    # Crystal Tower 2
+    z2 = UCpt(IMS, kind='normal', doc='LOM Xtal2 Z')
+    x2 = UCpt(IMS, kind='normal', doc='LOM Xtal2 X')
+    y2 = UCpt(IMS, kind='normal', doc='LOM Xtal2 Y')
+    th2 = UCpt(IMS, kind='normal', doc='LOM Xtal2 Theta')
+    ch2 = UCpt(IMS, kind='normal', doc='LOM Xtal2 Chi')
+    h2n = UCpt(IMS, kind='normal', doc='LOM Xtal2 Hn')
+    diode2 = UCpt(IMS, kind='normal', doc='LOM Xtal2 PIPS')
+    # Diagnostic Tower
+    dh = UCpt(IMS, kind='normal', doc='LOM Dia H')
+    dv = UCpt(IMS, kind='normal', doc='LOM Dia V')
+    dr = UCpt(IMS, kind='normal', doc='LOM Dia Theta')
+    df = UCpt(IMS, kind='normal', doc='LOM Dia Filter Wheel')
+    dd = UCpt(IMS, kind='normal', doc='LOM Dia PIPS')
+    yag_zoom = UCpt(IMS, kind='normal', doc='LOM Zoom')
+
+    t1_c_ref = Cpt(EpicsSignalRO, ':T1C:REF', kind='omitted',
+                   doc='Tower 1 Diamond crystal reflection')
+    t1_si_ref = Cpt(EpicsSignalRO, ':T1Si:REF', kind='omitted',
+                    doc='Tower 1 Silicon crystal reflection')
+    t2_c_ref = Cpt(EpicsSignalRO, ':T2C:REF', kind='omitted',
+                   doc='Tower 2 Diamond crystal reflection')
+    t2_si_ref = Cpt(EpicsSignalRO, ':T2Si:REF', kind='omitted',
+                    doc='Tower 2 Silicon crystal reflection')
 
     # QIcon for UX
     _icon = 'fa.share-alt-square'
 
-    tab_whitelist = ['h1n', 'yag', 'dectris', 'diode', 'foil', 'remove_dia']
+    tab_whitelist = ['h1n', 'yag', 'dectris', 'diode', 'foil',
+                     'remove_dia']
 
     def __init__(self, prefix, *, name, main_line='MAIN', mono_line='MONO',
                  **kwargs):
+        kwargs['z1_prefix'] = kwargs.get('z1_prefix') or get_prefix('z1')
```

Comment:
I don't know if I should keep this way of defaulting to some pre-defined prefixes? 
Are they changing based on XPP and XCS?
Maybe storing them somewhere else not in this module might be better - if we go this way?
Also, this method does not make it very clear to the user that they are even going to be using these prefixes when creating this device.

Reply:
XCS does have a different set of prefixes, but otherwise it's the same device

I do think defaulting to the correct prefixes is good

Do we need to allow the user to change these at all? Could we just fully hard-code prefixes for the XPP and XCS LODCMs? There must be a concise way to define this... it's escaping me at the moment

Reply:
That would be great.

Is the prefix for the XCS motors: `HFX` ? (based on old code)
https://github.com/pcdshub/old-hutch-configs/blob/18c7769d66a647ab558ad7ebc34d019fc98fbf48/xcs/xcsmotors.py#L527-L554

If these are the pvs from the link above, there is only  one (i think) PV that differs from XPP:
```
self.lom_df = psmotor("XPP:MON:MMS:27", "lom_df")

self.lom_df = psmotor("HFX:MON:MMS:22", "lom_df")  # calib foils
```

Reply:
That looks correct to me. `HFX` a prefix used for some items in the tunnel between `XPP` and `XCS`, and since the mono is in there the PV name makes sense.

It's very lucky if only one of the number indices is different. I wonder why the engineer changed exactly one... maybe the `XPP` mono motor prefixes include the CCM motors? `XPP` has two monochromators right next to each other... Maybe the foil stage was added later? I can only guess.